test/extensions/filters/http/wasm/test_data/test_grpc_call_cpp.cc

```diff
@@ -80,4 +80,40 @@ FilterHeadersStatus GrpcCallContext::onRequestHeaders(uint32_t, bool end_of_stre
   return FilterHeadersStatus::StopIteration;
 }
 
+class GrpcCallContext : public Context {
+public:
+  explicit GrpcCallContext(uint32_t id, RootContext* root) : Context(id, root) {}
+
+  FilterHeadersStatus onRequestHeaders(uint32_t, bool) override;
+
+  GrpcCallRootContext* root() { return static_cast<GrpcCallRootContext*>(Context::root()); }
+};
+
+static RegisterContextFactory register_GrpcCallFallbackContext(CONTEXT_FACTORY(GrpcCallContext),
+                                                       ROOT_FACTORY(GrpcCallRootContext),
+                                                       "grpc_call");
+
+FilterHeadersStatus GrpcCallContext::onRequestHeaders(uint32_t, bool end_of_stream) {
+  google::protobuf::Value value;
+  value.set_string_value("request");
+  HeaderStringPairs initial_metadata;
+  root()->handler_ = new MyGrpcCallHandler();
+  if (end_of_stream) {
+    if (root()->grpcCallHandler("cluster", "service", "method", initial_metadata, value,
+                                1000, std::unique_ptr<GrpcCallHandlerBase>(root()->handler_)) ==
+        WasmResult::Ok) {
+      logError("expected failure did not occur");
+    }
+    return FilterHeadersStatus::Continue;
+  }
+  root()->grpcCallHandler("cluster", "service", "method", initial_metadata, value, 1000,
+                          std::unique_ptr<GrpcCallHandlerBase>(root()->handler_));
+  if (root()->grpcCallHandler(
+          "bogus grpc_service", "service", "method", initial_metadata, value, 1000,
+          std::unique_ptr<GrpcCallHandlerBase>(new MyGrpcCallHandler())) == WasmResult::Ok) {
+    logError("bogus grpc_service accepted error");
+  }
```

Comment:
I know the existing tests are full of this pattern, but this construction is wrong, since it doesn't test if the test case passed or even if it was executed at all.

This should be changed to positive tests, i.e.:
```
if (root()->grpcCallHandler(...) == WasmResult::BadArgument) {
  logInfo("bogus grpc_service rejected");
}
```
and the test runner should verify that this log entry was generated by the plugin.

Same in other new tests.

Reply:
done